### PR TITLE
[metadata manager m8s] [Validator] implementation for validator

### PR DESCRIPTION
**Pull request by josephjacobmorris**
**What type of PR is this?**
> /kind feature

**What this PR does / why we need it**:
This PR helps in validating different query params . and  gives appropriate message to the user if it is invalid. 

**Test Report Added?**:
> /kind TESTED

**Test Report**:
* Valid Scenario
image

* If region is invalid and type is given as AWS-S3
image

* If SizeOfBucket is invalid
image

* if Size Operator is invalid
image

**Special notes for your reviewer**:

### CODE REVIEW

File: metadata/pkg/service/service.go [hunk 1]
```diff
@@ -157,7 +158,18 @@ func (f *metadataService) SyncMetadata(ctx context.Context, in *pb.SyncMetadataR
 }
 
 func (f *metadataService) ListMetadata(ctx context.Context, in *pb.ListMetadataRequest, out *pb.ListMetadataResponse) error {
-	log.Info("Received GetMetadata request in metadata service.")
+	log.Info("received GetMetadata request in metadata service.")
```
Comment (PravinRanjan10) [inline metadata/pkg/service/service.go:161]: plz be cosistent with either starting char should be caps or small.

File: metadata/pkg/validator/AWSQueryValidator.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package validator
```
Comment (PravinRanjan10) [inline metadata/pkg/validator/AWSQueryValidator.go:1]: just to validate region, why do we need seprate file.

Comment (PravinRanjan10) [inline metadata/pkg/validator/AWSQueryValidator.go:1]: just to validate region, why do we need seprate file.

File: metadata/pkg/validator/generalQueryValidator.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+	case constants.LESS_THAN_EQUAL_OPERATOR:
+
+	case constants.LESS_THAN_OPERATOR:
+
+	case constants.GREATER_THAN_EQUAL_OPERATOR:
+
+	case constants.GREATER_THAN_OPERATOR:
+		// operator is valid
+		break
```
Comment (PravinRanjan10) [inline metadata/pkg/validator/generalQueryValidator.go:38]: why do we need break here?

File: metadata/pkg/constants/Constants.go [hunk 4]
```diff
@@ -0,0 +1,36 @@
+package constants
+
+const (
+	CACHE_NAME = "cachename"
+
```
Comment (PravinRanjan10) [inline metadata/pkg/constants/Constants.go:5]: you may remove gap of line between two lines.

**Review by PravinRanjan10 — CHANGES_REQUESTED** [review]

File: metadata/pkg/service/service.go (hunk 1 above)
Comment (josephjacobmorris) [reply metadata/pkg/service/service.go]: keeping all info logs in lower case 
keeping all error logs starting with uppercase

File: metadata/pkg/validator/AWSQueryValidator.go (hunk 2 above)
Comment (josephjacobmorris) [reply metadata/pkg/validator/AWSQueryValidator.go]: moved all validation logic to validator file

File: metadata/pkg/validator/generalQueryValidator.go (hunk 3 above)
Comment (josephjacobmorris) [reply metadata/pkg/validator/generalQueryValidator.go]: moved all validation logic to validator file

File: metadata/pkg/constants/Constants.go (hunk 4 above)
Comment (josephjacobmorris) [reply metadata/pkg/constants/Constants.go]: done

File: metadata/pkg/validator/AWSQueryValidator.go (hunk 2 above)
Comment (josephjacobmorris) [reply metadata/pkg/validator/AWSQueryValidator.go]: moved all validation logic to validator file

**Review by PravinRanjan10 — APPROVED** [review]
LGTM

**Pull request merged**